homeassistant/components/notify/joaoapps_join.py

```diff
@@ -0,0 +1,75 @@
+"""
+Join platform for notify component.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/notify.Join/
+"""
+import logging
+import voluptuous as vol
+from homeassistant.components.notify import (
+    ATTR_DATA, ATTR_TITLE, BaseNotificationService)
+from homeassistant.const import CONF_PLATFORM, CONF_NAME, CONF_API_KEY
+import homeassistant.helpers.config_validation as cv
+
+REQUIREMENTS = [
+    'https://github.com/nkgilley/python-join-api/archive/'
+    'ceb384eb21e2b103fc0c355447252fedd7f7a185.zip#python-join-api==0.0.1']
+
+_LOGGER = logging.getLogger(__name__)
+
+CONF_DEVICE_ID = 'device_id'
+
+PLATFORM_SCHEMA = vol.Schema({
+    vol.Required(CONF_PLATFORM): 'joaoapps_join',
+    vol.Required(CONF_DEVICE_ID): cv.string,
+    vol.Optional(CONF_NAME): cv.string,
+    vol.Optional(CONF_API_KEY): cv.string
+})
+
+
+# pylint: disable=unused-variable
+def get_service(hass, config):
+    """Get the Join notification service."""
+    device_id = config.get(CONF_DEVICE_ID)
+    api_key = config.get(CONF_API_KEY)
+    return JoinNotificationService(device_id, api_key)
+
+
+# pylint: disable=too-few-public-methods
+class JoinNotificationService(BaseNotificationService):
+    """Implement the notification service for Join."""
+
+    def __init__(self, device_id, api_key=None):
+        """Initialize the service."""
+        self._device_id = device_id
+        self._api_key = api_key
+        if self._api_key:
+            from pyjoin import get_devices
+            if not get_devices(self._api_key):
+                _LOGGER.error("Error connecting to Join, check API key")
```

Comment:
This only prints an error, it does not stop the platform from being initialized (you're still returning the object on line 35)